scripts/lib-carousel.js

```diff
@@ -0,0 +1,389 @@
+  activateBullet(`slider-dot-${activeSlide}`);
+};
+
+/**
+ * Navigate to previous slide
+ */
+const toPrevSlide = () => {
+  activeSlide = incrementActiveSlide(-1);
+  activateSlide(`slider-slide-${activeSlide}`);
+  activateBullet(`slider-dot-${activeSlide}`);
+};
+
+/**
+ * Start slide show
+ */
+const startSlideShow = () => {
+  if (!slideShow) {
+    slideShow = window.setInterval(() => toNextSlide(slideCount), sliderDurationMs);
+  }
+};
+
+/**
+ * Stop automatic Slideshow
+ */
+const stopSlideShow = () => {
+  window.clearInterval(slideShow);
+  slideShow = false;
+};
+
+/**
+ * Handle touch gestures (back and forward swipe) for touch devices
+ */
+function handleGesture() {
+  if (touchEndX < touchStartX) {
+    toNextSlide();
+  }
+  if (touchEndX > touchStartX) {
+    toPrevSlide();
+  }
+}
+
+/**
+ * Prevent scrolling when swiping for touch devices
+ * @param e event
+ */
+function disableScrolling(e) {
+  const isLink = e.target.tagName.toLowerCase() === 'a';
+  const isButton = e.target.tagName.toLowerCase() === 'button';
+  if (!isLink && !isButton) {
+    e.preventDefault();
+  }
+}
+
+/**
+ * Touch Move Event Listener
+ * @param slideContainer: slider wrapper or block containing the slides
+ */
+function touchMoveEl(slideContainer) {
+  slideContainer.addEventListener(
+    'touchmove',
+    (e) => {
+      disableScrolling(e);
+      touchRelX = Math.floor(e.touches[0].clientX) - touchStartX;
+      const slide = document.getElementById(`slider-slide-${activeSlide}`);
+      slide.style.left = `${touchRelX}px`;
+    },
+    { passive: false },
+  );
+}
+
+/**
+ * Touch Start Event Listener
+ * @param slideContainer: slider wrapper or block containing the slides
+ */
+function touchStartEl(slideContainer) {
+  slideContainer.addEventListener(
+    'touchstart',
+    (e) => {
+      disableScrolling(e);
+      touchStartX = Math.floor(e.touches[0].clientX);
+      stopSlideShow();
+    },
+    { passive: false },
+  );
+}
+
+/**
+ * Touch End Event Listener
+ * @param slideContainer: slider wrapper or block containing the slides
+ */
+function touchEndEl(slideContainer) {
+  slideContainer.addEventListener(
+    'touchend',
+    (e) => {
+      disableScrolling(e);
+      touchEndX = Math.floor(e.changedTouches[0].clientX);
+      handleGesture();
+    },
+    { passive: false },
+  );
+}
+
+/**
+ * Initialize Slider
+ * @param slidesLength: number of slides
+ */
+export function initSlider(slidesLength) {
+  slideCount = slidesLength;
+  const sliderWrapper = document.querySelector('.slider-wrapper');
+  sliderWrapper.addEventListener('mouseover', stopSlideShow);
+  sliderWrapper.addEventListener('mouseleave', startSlideShow);
+  touchStartEl(sliderWrapper);
+  touchMoveEl(sliderWrapper);
+  touchEndEl(sliderWrapper);
+  startSlideShow();
+  dottedNavOnClickEvents();
+  arrowNavOnClickEvents();
+}
+
+/**
+ * Create Arrow navigation
+ * @returns {HTMLDivElement}
+ */
+export function createArrowNav() {
```

Comment:
I have recently come across ["reflow"](https://dev.to/gopal1996/understanding-reflow-and-repaint-in-the-browser-1jbg) a browser has to do when changing elements. This translates to us having to try limiting DOM operations. This can often be achieved by reducing operations by setting HTML directly:

```
export function createArrowNav() {
  const arrowNavContainer = document.createElement('div');
  arrowNavContainer.classList.add('arrow-nav');

  const arrowLeft = `<button id="slider-arrow-left" aria-label="Previous Slide">${HTML_ARROW_LEFT}</button>`;
  const arrowRight = `<button id="slider-arrow-right" aria-label="Next Slide" role="button">${HTML_ARROW_RIGHT}</button>`;
  
  arrowNavContainer.innerHTML = arrowLeft + arrowRight;
  
  return arrowNavContainer;
}
```